internal/api/service.go

```diff
@@ -0,0 +1,40 @@
+package api
+
+import (
+	"context"
+	"github.com/rs/zerolog"
+	"google.golang.org/protobuf/types/known/emptypb"
+	api "ova-serial-api/pkg/ova-serial-api"
+)
+
+type OvaSerialAPI struct {
+	api.UnimplementedOvaSerialV1Server
+	logger *zerolog.Logger
+}
+
+func NewSerialAPI(logger *zerolog.Logger) api.OvaSerialV1Server {
+	return &OvaSerialAPI{
+		UnimplementedOvaSerialV1Server: api.UnimplementedOvaSerialV1Server{},
+		logger:                         logger,
+	}
+}
+
+func (a *OvaSerialAPI) CreateSerialV1(ctx context.Context, req *api.CreateSerialRequestV1) (*api.CreateSerialResponseV1, error) {
+	a.logger.Info().Msgf("CreateSerial request: %v", req)
+	return a.UnimplementedOvaSerialV1Server.CreateSerial(ctx, req)
```

Comment:
Я думаю, в этих методах не нужно вызывать методы UnimplementedOvaSerialV1Server, потому что они просто генерируют ошибки. Лучше 
```suggestion
	return nil, nil
```
И для остальных – тоже